app/services/EmploymentSessionService.scala

```diff
@@ -60,6 +60,15 @@ class EmploymentSessionService @Inject()(employmentUserDataRepository: Employmen
     employmentUserDataRepository.find(taxYear, employmentId)
   }
 
+  def getSessionDataAndReturnResult(taxYear: Int, employmentId: String)(result: EmploymentUserData => Future[Result])
+                                   (implicit user: User[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
+
+    employmentUserDataRepository.find(taxYear, employmentId).flatMap {
+      case Some(employmentUserData: EmploymentUserData) => result(employmentUserData)
+      case None => Future(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
```

Comment:
do you want a logger in here to say anything about not finding data and redirecting?

Reply:
the find method already logs it